src/coingecko_pricing.py

```diff
@@ -0,0 +1,90 @@
+import os
+from typing import Optional
+import requests
+import json
+from web3 import Web3
+from src.config import logger
+from src.constants import NATIVE_ETH_TOKEN_ADDRESS
+
+coingecko_api_key = os.getenv("COINGECKO_API_KEY")
+
+
+def load_cleaned_token_list(file_path):
+    with open(file_path, "r") as f:
+        return json.load(f)
+
+
+def get_token_id_by_address(cleaned_token_list, token_address):
+    for token in cleaned_token_list:
+        if token["platforms"].get("ethereum") == token_address:
+            return token["id"]
+    return None
+
+
+def get_api_price(
+    token_id: str, start_timestamp: int, end_timestamp: int
+) -> Optional[float]:
+    """
+    Makes call to Coingecko API to fetch price, between a start and end timestamp.
+    """
+    if not coingecko_api_key:
+        logger.warning("Coingecko API key is not set.")
+        return None
+    # price of token is returned in ETH
+    url = (
+        f"https://pro-api.coingecko.com/api/v3/coins/{token_id}/market_chart/range"
+        f"?vs_currency=eth&from={start_timestamp}&to={end_timestamp}"
+    )
+    headers = {
+        "accept": "application/json",
+        "x-cg-pro-api-key": coingecko_api_key,
+    }
+    try:
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        data = response.json()
+        # return available coingecko price, which is the closest to the block timestamp
+        if len(data["prices"]) != 0:
+            price = data["prices"][0][1]
+            return price
+        return None
+    except requests.RequestException as e:
+        logger.warning(f"Error fetching price from Coingecko API: {e}")
+        return None
+
+
+def get_price(web3: Web3, block_number: int, token_address: str, tx_hash: str):
+    """
+    Function returns coingecko price for a token address,
+    closest to block timestamp for a given tx hash.
+    """
+    cleaned_token_list = load_cleaned_token_list(
```

Comment:
Isn't it wasteful to keep reading the file again and again each time we want to get a price feed? Why not load it in a variable once and be done with it?